src/Morphir/Spark/AST.elm

```diff
@@ -157,6 +159,28 @@ objectExpressionFromValue ir morphirValue =
         Value.Variable _ varName ->
             From varName |> Ok
 
+        Value.Apply _ ((Value.Lambda _ _ (Value.List _ [ Value.Record _ _ ])) as lambda) sourceRelation ->
+            -- e.g. source |> List.map .fieldName |> (\a -> [{min = List.minimum(a)}])
+            objectExpressionFromValue ir sourceRelation
+                |> Result.andThen
+                    (\sourceExpression ->
+                        namedExpressionsFromValue ir lambda
+                            |> Result.andThen
+                                (\lambdaExpressions ->
+                                    case sourceExpression of
+                                        Select (( _, subExpression ) :: []) oldSourceExpression ->
+                                            case lambdaExpressions of
+                                                ( lamName, Function fname [ a ] ) :: [] ->
+                                                    Select [ ( lamName, Function fname [ subExpression ] ) ] oldSourceExpression |> Ok
```

Comment:
I'm not sure if `fname` refers to a spark function, because a `Function` name gets prefixed with `org.apache.spark.sql.functions` in the Backend.elm.

Reply:
It currently doesn't seem like there's a better way to refer to a user defined function or package library function. My suggestion would be to change the signature of `Function String (List Expression)` to something like `Function ModulePath String (List Expression)` where ModulePath is type alias like `type alias ModulePath = List String`. This would make `Function` reusable.

Reply:
I'm expecting all the functions that get used here to be spark functions that are prefixed with `org.apache.spark.sql.functions`. I'm not aware of any other spark modules we're expecting to call functions from yet, and as I understand it, our user-defined functions all get inlined, too.

I'm not sure why you're suggesting to make Function able to take other module paths here. Is there something I've misunderstood?